crates/ibc/src/core/ics04_channel/handler/timeout.rs

```diff
@@ -133,9 +134,21 @@ pub fn process<Ctx: ChannelReader>(
 
     output.log("success: packet timeout ");
 
-    output.emit(IbcEvent::TimeoutPacket(TimeoutPacket {
-        packet: packet.clone(),
-    }));
+    output.emit(IbcEvent::TimeoutPacket(TimeoutPacket::new(
+        packet.clone(),
+        source_channel_end.ordering,
+    )));
+
+    if source_channel_end.order_matches(&Order::Ordered) {
+        output.emit(IbcEvent::ChannelClosed(ChannelClosed::new(
```

Comment:
Also, we're not handling the noop case here and in the timeoutOnClose handler too. (unrelated to this PR) -> https://github.com/cosmos/ibc-go/blob/a4f7761981771b453fdb06203dba33a62b4b6317/modules/core/04-channel/keeper/timeout.go#L77-L84
https://github.com/cosmos/ibc-go/blob/a4f7761981771b453fdb06203dba33a62b4b6317/modules/core/04-channel/keeper/timeout.go#L226-L233

Reply:
created #213